[com_media] Refactoring of thumbnails and details view

jissereitsma opened the pull request:
Part of the refactoring work of @Buddhima (https://github.com/joomla/joomla-cms/pull/3839) was to add better delete-actions and edit-actions for images. While working on updating his during a code sprint, we found that the current thumbnails and details view was lacking flexibility. This PR includes refactoring to allow for more flexible updating in future work.

This PR adds first of all JLayout files for delete() and edit() actions. Because an edit() action is currently missing (it is part of the bundled works of @Buddhima though), there is a simple check to see whether the "image" view is there or not. The delete() action has been refactored to use the Delete toolbar-button, instead of using its own link. This way, future checks in that Delete toolbar-button ("Hey, you did not select anything") also applies to these actions.

Because of these changes, the HTML has changed in various places as well: For instance, the icons have been made a little bit bigger. In future PRs, this code needs beautifying: Removing style-arguments with proper CSS rules, new CSS-rules for the thumbnail-sizes, etcetera.

For testing, apply the patch, and try to delete the various files in either Thumnails-view or Details-view: Use all types of files: Images, ZIP-files, PDF-files. Try to delete an empty folder as well. And try to delete a non-empty folder - because it should not work ;)

jissereitsma committed:
Add edit icons for editing image + popup preview

jissereitsma committed:
New JLayout files for reusage in media manager

jissereitsma committed:
Rewrite thumbnail fails for usage with new JLayouts

jissereitsma committed:
Better way of using JLayouts

jissereitsma committed:
Code formatting

jissereitsma committed:
New JLayoutFile for delete-actions in detail view

jissereitsma committed:
Implement JLayout for delete-action

jissereitsma committed:
Run delete function via form button "Delete" instead of direct link

jissereitsma committed:
Cleanup after implementing delete-task via Delete-button

jissereitsma committed:
Implement delete-task via Delete-button for thumbnail-views

jissereitsma committed:
Update details-view to look same as regular (article) listing

jissereitsma committed:
Merge remote-tracking branch 'upstream/3.5-dev' into new_com_media_jisse_01

Conflicts:
	administrator/components/com_media/views/medialist/tmpl/details_doc.php
	administrator/components/com_media/views/medialist/tmpl/details_folder.php
	administrator/components/com_media/views/medialist/tmpl/details_img.php
	administrator/components/com_media/views/medialist/tmpl/details_up.php
	administrator/components/com_media/views/medialist/tmpl/thumbs_doc.php
	administrator/components/com_media/views/medialist/tmpl/thumbs_folder.php
	administrator/components/com_media/views/medialist/tmpl/thumbs_img.php
	administrator/components/com_media/views/medialist/tmpl/thumbs_up.php

jissereitsma committed:
Fix merge issue

jissereitsma committed:
Fix merge conflicts

jissereitsma committed:
Fix merge issues + JS error because of changes core.js

wilsonge commented:
You appear to have added an empty file? (https://github.com/jissereitsma/joomla-cms/blob/new_com_media_jisse_01/administrator/components/com_media/controllers/image.php)

jissereitsma committed:
Remove dummy file

jissereitsma commented:
@wilsonge Correct. During the code sprint, we started to implement a new MVC class-set for "image" as well. But later on, we decided to focus on some smaller tasks instead. I'll remove the file again.

zero-24 committed:
CS

zero-24 committed:
let travis be happy ;)

zero-24 committed:
Update details_img.php

Kubik-Rubik commented:
@jissereitsma Could you please merge @zero-24's PR against your branch so that we make Travis happy? Thanks! :-)

jissereitsma committed:
Merge pull request #2 from zero-24/patch-6

Fix Travis CS errors

jissereitsma commented:
Many thanks, @zero-24 @Kubik-Rubik. The PR is merged in my branch.

jissereitsma committed:
Fix broken JHtml image tag

wilsonge commented on administrator/components/com_media/views/medialist/tmpl/details.php:
```diff
@@ -25,41 +26,54 @@
 </form>
+
+<script>
+	document.adminForm = document['mediamanager-form'];
```
Please put this in the head using `JDocument::addScriptDeclaration` before @dgt41 gets the shotgun out :)

jissereitsma replied:
I ... don't ... know ... why ... this ... happened. Fixed.

peterlose commented:
I'm getting some notices:

![skaermbillede 2015-08-25 kl 10 36 22](https://cloud.githubusercontent.com/assets/1738811/9462310/2e076ab4-4b15-11e5-8bf3-759486d40144.png)

jissereitsma committed:
Update details.php

Move script from button to top, now using addScriptDeclaration

jissereitsma committed:
Remove PHP notices on $width_150 and $height_150

Get rid of strange uninitialized image proportions. The code originates from `models/list.php` which uses some extreme logic to create the right width/height for various sizes, regardless of whether these sizes are actually needed. A better approach would be to simply generate the width and height of the image through a simple helper-function. For later worry.

wilsonge commented:
@test I just had a play around with this and it seems like a good improvement to me :) The view in com_media was significantly better. No apparent issues using the image editor-xtd button either in frontend or backend editing.

My only slight issue was that I felt using the preview button wasn't 100% obvious as it was a relatively small icon. I would reintroduce clicking on the image also opening the preview to make things easy for users

Worti2 commented:
The Patch is not OK.
- In the thumbnail-view and the detail-view of the backend the delete-button next to create-new-folder-button is without function.
- If you select more than one picture in the thumbnail-view and if you click on the red x of one picture, you will delete all chosen (I do not know if this is an issue).
- In the detail-view it is not possible to choose all pictures with the check-all checkbox.
This is OK:
- You are not able to delete a folder if there are still pictures in.
Regards
Markus

Worti2 commented:
I have tested this item :red_circle: unsuccessfully on e6d05a6aec9814235e1811322b04beee37cdbafb<br /><br />The Patch is not OK.
- In the thumbnail-view and the detail-view of the backend the delete-button next to create-new-folder-button is without function.
- If you select more than one picture in the thumbnail-view and if you click on the red x of one picture, you will delete all chosen (I do not know if this is an issue).
- In the detail-view it is not possible to choose all pictures with the check-all checkbox.
This is OK:
- You are not able to delete a folder if there are still pictures in.
Regards Markus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>

roland-d commented:
@jissereitsma Is this obsolete with the new work you are doing on the media manager?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>

yireo commented:
Yes, it is. Can you close this PR?

brianteeman commented:
Closed as requested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7646">issues.joomla.org/joomla-cms/7646</a>.</sub>
